manifests/init.pp

```diff
@@ -491,7 +491,7 @@
   Optional[Enum['On', 'Off', 'on', 'off']] $ldap_verify_server_cert     = undef,
   Optional[String] $ldap_trusted_mode                                   = undef,
   Boolean $error_documents                                              = false,
-  String $timeout                                                       = '60',
+  Integer $timeout                                                      = 60,
```

Comment:
IMHO an integer would be better, but it is a breaking change. Would it be better to make it
```suggestion
  Variant[Integer[0], String[1]] $timeout                                                      = 60,
```

(and then properly align it, that's hard in GH)

Reply:
What do you think of `Variant[Integer[0], Pattern[/^[0-9]+$/]]`?

Reply:
FYI: The next release will be a major one, so if we want to break things, now is a good time

Reply:
> FYI: The next release will be a major one, so if we want to break things, now is a good time

Adding Datatypes to the Parameters actually was a good Idea, but IMHO it was done only half-hearted.  This should be improved for the next Major-Release.